lib/resource/resource.ex

```diff
@@ -2093,24 +2103,28 @@ defmodule AshGraphql.Resource do
   end
 
   def type_definition(resource, api, schema) do
-    type = Resource.type(resource)
+    if generate_object?(resource) do
+      type = Resource.type(resource)
 
-    interfaces =
-      if relay?(resource) do
-        [:node]
-      else
-        []
-      end
+      interfaces =
+        if relay?(resource) do
+          [:node]
+        else
+          []
+        end
 
-    %Absinthe.Blueprint.Schema.ObjectTypeDefinition{
-      description: Ash.Resource.Info.description(resource),
-      interfaces: interfaces,
-      fields: fields(resource, api, schema),
-      identifier: type,
-      module: schema,
-      name: Macro.camelize(to_string(type)),
-      __reference__: ref(__ENV__)
-    }
+      %Absinthe.Blueprint.Schema.ObjectTypeDefinition{
+        description: Ash.Resource.Info.description(resource),
+        interfaces: interfaces,
+        fields: fields(resource, api, schema),
+        identifier: type,
+        module: schema,
+        name: Macro.camelize(to_string(type)),
+        __reference__: ref(__ENV__)
+      }
+    else
+      []
```

Comment:
can we return `nil` here instead? Otherwise, looks great!

Reply:
done